clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBasedStoreSchemaFetcher.java

```diff
@@ -74,6 +76,21 @@ public Schema getLatestValueSchema() {
     }
   }
 
+  @Override
+  public List<Schema> getAllValueSchemas() {
+    String valueSchemaRequestPath = TYPE_VALUE_SCHEMA + "/" + storeClient.getStoreName();
+    MultiSchemaResponse multiSchemaResponse = fetchAllValueSchemas(valueSchemaRequestPath);
+    List<Schema> schemaList = new ArrayList<>();
+    try {
+      for (MultiSchemaResponse.Schema schema: multiSchemaResponse.getSchemas()) {
+        schemaList.add(parseSchemaFromJSONLooseValidation(schema.getSchemaStr()));
```

Comment:
IMU, the API expects the schemas would be ordered by schema id. I believe `TYPE_VALUE_SCHEMA + "/" + storeClient.getStoreName()` API call does not return the schemas in a sorted manner

Reply:
Hmm, actually, the requirement of this API only want to get a list of schema. 
Fedex needs it during start up time to iterate over all value schemas and compare and figure out which schema the beam record is using.
(Also, schema ID to me is an internal details, so we ideally should not expose this.)

Reply:
It would be better to clarify the requirements, and also consider what we'd like to offer to users via the API. Having an API that returns a randomly ordered list doesn't help IMO. Let's change it to `Set`/`Collection` if we plan on not having it be ordered. A `List` is inherently ordered